controllers/infrastructure/byomachine_controller.go

```diff
@@ -613,9 +613,16 @@ func (r *ByoMachineReconciler) markHostForCleanup(ctx context.Context, machineSc
 	}
 	machineScope.ByoHost.Annotations[infrav1.HostCleanupAnnotation] = ""
 
-	// remove the pf9 cluster label
-	logger.Info("Removing pf9 cluster label %s from ByoHost %s", infrav1.ClusterLabel, machineScope.ByoHost.Name)
-	delete(machineScope.ByoHost.Labels, infrav1.ClusterLabel)
+	// Debug: Log the value and presence of the upgrade-in-progress annotation
+	upgradeInProgress, ok := machineScope.ByoMachine.Annotations["barista.platform9.io/upgrade-in-progress"]
```

Comment:
Do we clear this annotation `machineScope.ByoMachine.Annotations["barista.platform9.io/upgrade-in-progress"]` when the upgrade is complete : either successfully or it fails?